Sources/UI/Channels/ChannelTableView.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  ChannelTableView.swift
+//  StreamChat
+//
+//  Created by Bahadir Oncel on 06/03/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+
+/// `UITableView` subclass used for keeping track of registered `ChannelTableViewCell` subclasses..
+///
+/// Registered cell info are used to automatically dequeue them as correct type in `ChannelsViewController`.
+class ChannelTableView: UITableView {
+    var registeredClass: (identifier: String, `class`: AnyClass) = (identifier: ChannelTableViewCell.reuseIdentifier, `class`: ChannelTableViewCell.self)
```

Comment:
I suggest to move it to `TableView` with:
```suggestion
    var interestedRegisteredClasses = [String: AnyClass]()
```

In `ChannelsViewController` it will be:
```swift
    var interestedRegisteredClasses = [ChannelTableViewCell.reuseIdentifier: ChannelTableViewCell.self]
```